activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -1,8 +1,18 @@
 # frozen_string_literal: true
 
 require "active_support/core_ext/numeric/bytes"
-require "azure/storage"
-require "azure/storage/core/auth/shared_access_signature"
+
+begin
+  require "azure/storage"
+  require "azure/storage/core/auth/shared_access_signature"
+rescue LoadError
+  raise(
+    ActiveStorage::MissingAdapterError,
+    "Add the azure-storage gem to your Gemfile " \
+    "in order to use Asure Storage service:\n\n" \
```

Comment:
I also think these should say "in order to use _the_ Cloud Name service, but that's mostly personal preference.

```suggestion
    "in order to use Azure Storage service:\n\n" \
```